File: petproject/aws/backend-jvm/src/main/java/com/cramsan/petproject/awslambda/dao/PlantMainNameImpl.java
package com.cramsan.petproject.awslambda.dao;

import com.cramsan.petproject.appcore.storage.PlantMainName;

import software.amazon.awssdk.enhanced.dynamodb.mapper.annotations.DynamoDbBean;
import software.amazon.awssdk.enhanced.dynamodb.mapper.annotations.DynamoDbPartitionKey;

@DynamoDbBean
public class PlantMainNameImpl implements PlantMainName {
    private long id;
    private String MainName;
    private long plantId;
    private String locale;

    public PlantMainNameImpl() { }

    public PlantMainNameImpl(long id, String MainName, long plantId, String locale) {
        this.id = id;
        this.MainName = MainName;
        this.plantId = plantId;
        this.locale = locale;
    }

    @DynamoDbPartitionKey
    @Override
    public long getId() {
        return id;
    }

    public void setId(long id) {
        this.id = id;
    }

    @Override
    public String getMainName() {
        return MainName;
    }

    public void setMainName(String MainName) {
        this.MainName = MainName;
    }

    @Override
    public long getPlantId() {
        return plantId;
    }

    public void setPlantId(long plantId) {
        this.plantId = plantId;
    }

    @Override
    public String getLocale() {
        return locale;
    }

    public void setLocale(String locale) {
        this.locale = locale;
    }
}

File: petproject/awslambda-java/src/main/kotlin/com/cramsan/petproject/awslambda/dao/PlantMainNameImpl.kt
package com.cramsan.petproject.awslambda.dao

import com.cramsan.petproject.appcore.storage.PlantMainName
import software.amazon.awssdk.enhanced.dynamodb.mapper.annotations.DynamoDbAttribute
import software.amazon.awssdk.enhanced.dynamodb.mapper.annotations.DynamoDbBean
import software.amazon.awssdk.enhanced.dynamodb.mapper.annotations.DynamoDbPartitionKey

@DynamoDbBean
class PlantMainNameImpl() : PlantMainName {
    @get:DynamoDbPartitionKey
    @get:DynamoDbAttribute(value = "id")
    var _id: Long = -1
    @get:DynamoDbAttribute(value = "mainName")
    lateinit var _mainName: String
    @get:DynamoDbAttribute(value = "plantId")
    var _plantId: Long = -1
    @get:DynamoDbAttribute(value = "locale")
    lateinit var _locale: String

    constructor(
        id: Long,
        mainName: String,
        plantId: Long,
        locale: String,
    ) : this() {
        _id = id
        _mainName = mainName
        _plantId = plantId
        _locale = locale
    }

    override val id: Long = _id
    override val mainName: String = _mainName
    override val plantId: Long = _plantId
    override val locale: String = _locale
}
